cpp/src/arrow/ipc/reader.cc

```diff
@@ -1240,6 +1382,14 @@ class RecordBatchFileReaderImpl : public RecordBatchFileReader {
     return std::move(message);
   }
 
+  Result<std::unique_ptr<Message>> ReadCachedMessageFromBlock(const FileBlock& block) {
+    ++stats_.num_messages;
+    ARROW_ASSIGN_OR_RAISE(std::shared_ptr<Buffer> metadata,
+                          metadata_cache_->Read({block.offset, block.metadata_length}));
+    std::shared_ptr<Buffer> data;
+    return arrow::ipc::ReadMessageFromCached(std::move(metadata), std::move(data));
+  }
+
```

Comment:
I could go either way but it feels like this would be a little clearer inlined into its only callsite above

Reply:
Inlined.